### DISCUSSION TITLE
API de consulta da tabela FIPE

### DISCUSSION THREAD

**Discussion by tetri** [BrasilAPI/BrasilAPI#238] [Ideas]
Já que estamos falando em Brasil e dados públicos, que tal adicionar rotas de consulta à Tabela FIPE.
Sei que já existe muitas API e lugares para consulta desses dados, mas seria mais uma rota bem útil para este projeto.

**Comment by michaelLean** [comment]
Existe essa API que disponibiliza os dados da Fipe: https://fipeapi.appspot.com/, acredito com os dados dela seja fácil implementar as consultas no projeto, sem precisar fazer a mineração dos dados no site.

**Comment by brunofamiliar** [comment]
O @deividfortuna tem um trabalho incrível relacionado a esse assunto. Tem mais detalhes sobre no link: https://deividfortuna.github.io/fipe/. Acredito ser pertinente ter um endpoint nesse projeto que trata sobre essa questão, mesmo não estando relacionado a serviços de localização, consultas do gênero é bastante explorada por todos os brasileiros. Como o intuito é transformar o Brasil em uma grande api, vale a pena definirmos isso.

**Comment by deividfortuna** [comment]
Opa @brunofamiliar obrigado por me mencionar, vou dar uma olhada aqui como posso ajudar o projeto 😎

**Comment by deividfortuna** [comment]
Estou trabalhando em uma lib em js para depois adicionar na BrasilAPI: 
Repo: https://github.com/deividfortuna/fipe-promise
Progresso: https://github.com/deividfortuna/fipe-promise/issues/1

**Comment by brunofamiliar** [comment]
> Estou trabalhando em uma lib em js para depois adicionar na BrasilAPI:
> Repo: https://github.com/deividfortuna/fipe-promise
> Progresso: [deividfortuna/fipe-promise#1](https://github.com/deividfortuna/fipe-promise/issues/1)

Show! Vou acompanhar e vê se consigo contribuir também 😄

**Comment by lucianopf** [comment]
> Existe essa API que disponibiliza os dados da Fipe: https://fipeapi.appspot.com/, acredito com os dados dela seja fácil implementar as consultas no projeto, sem precisar fazer a mineração dos dados no site.

Poxa essa API parece estar fora do ar 😢 

Já essas do @deividfortuna usando a `parallelum` parece estar lindinha! 😱 🚀 

Bem interessante a ideia e o que está sendo desenvolvido em relação ao tema! 👏 🔝

**Comment by deividfortuna** [comment]
@lucianopf Muito obrigado pelo apoio, a biblioteca para consumir a Fipe API está quase pronta, vou voltar a trabalhar nesse PR no começo de 2021. Uma pergunta que ainda esta pendente é o tratamento de erros que ja foi citado no PR #116 
🚀

**Comment by calieno** [comment]
E aí David, como anda a API, estou pesquisando alguma coisa aqui também, pelo que vi a api da FIPE é paga por consulta, então creio que estaria violando alguma coisa em usa-la. Uma outra api interessante também procurada pelo mesmo público seria a consulta de placa.

**Comment by LorhanSohaky** [comment]
Nossa, só vi agora essa discussão. Acabei implementando em #301 . Imagino que vocês irão gostar... nela não é preciso especificar o tipo de veículo, ano modelo e demais informações que as outras APIs pedem

**Comment by g3luka** [comment]
@LorhanSohaky ficou sensacional os endpoint da FIPE, parabéns!
Já estou usando em um projeto de um cliente que é relacionado com automóveis de vários tipos e ajudou bastante.

Só senti falta de um endpoint que seria o que retorna todos os Modelos de uma determinada Marca e Tipo.
Como quando você vai procurar seu carro no site da FIPE, e seleciona primeiro o Tipo e depois a Marca, e ele lista abaixo todos os Modelos e Anos desta Marca+Tipo, saca?

Eu inspecionei o site da FIPE e pequei o endpoint que ele envia o TIPO, MARCA e TABELA DE REFERENCIA e retorna todos os MODELOS e ANOS: `https://veiculos.fipe.org.br/api/veiculos/ConsultarModelos`
E os dados necessários são: `codigoTipoVeiculo`, `codigoTabelaReferencia` e `codigoMarca`

Eu posso até fazer essa parte e enviar uma PR, mas é a minha primeira vez fazendo isso e fiquei com medo de fazer merda, então posso precisar de um help para fazer o merge na revisão.

Mais uma vez, parabéns pelo trabalho, esse projeto está ficando sensacional.
Abraço.
